sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala

```diff
@@ -211,6 +213,10 @@ class HiveExternalCatalogVersionsSuite extends SparkSubmitTestUtils {
         tryDownloadSpark(version, sparkTestingDir.getCanonicalPath)
       }
 
+      Files.deleteIfExists(Paths.get(sparkHome.getCanonicalPath, "jars", "ivy-2.5.1.jar"))
+      val ivyUrl = new URL("https://repo1.maven.org/maven2/org/apache/ivy/ivy/2.5.2/ivy-2.5.2.jar")
+      IOUtils.copy(ivyUrl, new File(s"${sparkHome.getCanonicalPath}/jars", "ivy-2.5.2.jar"))
```

Comment:
I have been investigating the case of the previous Ivy upgrade failure recently. I found that if the Spark version being tested in `HiveExternalCatalogVersionsSuite` also uses Ivy 2.5.2, the error like https://github.com/apache/spark/pull/44477#issuecomment-1868917258 will not occur again... But I haven't come up with a better solution yet.